src/data/validation/email.js

```diff
@@ -155,19 +183,48 @@ const returnValidatedEmails = (formData) => {
   if (errorsDict._error.length > 0) {
     throw new SubmissionError(errorsDict);
   }
-  let emails = [];
+  let emails = extractEmails(formData).allEmails;
+  emails = _.union(emails); // Dedup emails
+  return validateEmailAddresses(emails).validEmails;
+};
+
+// Combine all the rows from textarea and CSV and then make a map of email to salesforce id
+const parseData = (formData) => {
+  const rows = [];
+  const allRecords = {};
+
   if (formData[EMAIL_ADDRESS_TEXT_FORM_DATA] && formData[EMAIL_ADDRESS_TEXT_FORM_DATA].length) {
-    emails.push(...formData[EMAIL_ADDRESS_TEXT_FORM_DATA].split(/\r\n|\n/));
+    rows.push(...formData[EMAIL_ADDRESS_TEXT_FORM_DATA].split(/\r\n|\n/));
   }
+
   if (formData[EMAIL_ADDRESS_CSV_FORM_DATA] && formData[EMAIL_ADDRESS_CSV_FORM_DATA].length) {
-    emails.push(...formData[EMAIL_ADDRESS_CSV_FORM_DATA]);
+    rows.push(...formData[EMAIL_ADDRESS_CSV_FORM_DATA]);
   }
-  emails = _.union(emails); // Dedup emails
-  return validateEmailAddresses(emails).validEmails;
+
+  rows.forEach((row) => {
+    const [email, id] = row.split(',').map(item => item.trim());
+    allRecords[email] = id;
+  });
+
+  return allRecords;
+};
+
+// Extract salesforce ids for all validated emails
+const extractSalesforceIds = (formData, userEmails) => {
+  const parsedData = parseData(formData);
+  const ids = [];
+
+  userEmails.forEach((email) => {
+    ids.push(parsedData[email]);
+  });
+
+  const allFalse = ids.every(item => !item);
+  return allFalse ? undefined : ids;
```

Comment:
nit: perhaps add a comment and/or rename this variable to explain this logic a bit explicitly? It's not clear to me what `allFalse` really means in this context.

Reply:
this check ensure that if there are no salesforce ids present then do not send an empty array to backend.